pallets/pallet-bonded-coins/src/benchmarking.rs

```diff
@@ -1 +1,855 @@
+		// Verify
+		let (_, pool) = Pools::<T>::iter().next().expect("Pool should exist");
+		assert_eq!(pool.manager, None);
+	}
+
+	#[benchmark]
+	fn set_lock() {
+		let origin = T::DefaultOrigin::try_successful_origin().expect("creating origin should not fail");
+		let account_origin = origin
+			.clone()
+			.into_signer()
+			.expect("generating account_id from origin should not fail");
+		make_free_for_deposit::<T>(&account_origin);
+
+		let bonded_coin_id = T::BenchmarkHelper::calculate_bonded_asset_id(0);
+		create_bonded_asset::<T>(bonded_coin_id.clone());
+
+		let curve = get_linear_bonding_curve::<CurveParameterTypeOf<T>>();
+		let pool_id = create_pool::<T>(
+			curve,
+			[bonded_coin_id.clone()].to_vec(),
+			Some(account_origin.clone()),
+			None,
+			None,
+		);
+
+		#[extrinsic_call]
+		_(origin as T::RuntimeOrigin, pool_id, Locks::default());
+		// Verify
+		let (_, pool) = Pools::<T>::iter().next().expect("Pool should exist");
+		assert_eq!(pool.state, PoolStatus::Locked(Locks::default()));
+	}
+
+	#[benchmark]
+	fn unlock() {
+		let origin = T::DefaultOrigin::try_successful_origin().expect("creating origin should not fail");
+		let account_origin = origin
+			.clone()
+			.into_signer()
+			.expect("generating account_id from origin should not fail");
+		make_free_for_deposit::<T>(&account_origin);
+
+		let bonded_coin_id = T::BenchmarkHelper::calculate_bonded_asset_id(0);
+		create_bonded_asset::<T>(bonded_coin_id.clone());
+
+		let curve = get_linear_bonding_curve::<CurveParameterTypeOf<T>>();
+		let pool_id = create_pool::<T>(
+			curve,
+			[bonded_coin_id.clone()].to_vec(),
+			Some(account_origin.clone()),
+			Some(PoolStatus::Locked(Locks::default())),
+			None,
+		);
+
+		#[extrinsic_call]
+		_(origin as T::RuntimeOrigin, pool_id);
+		// Verify
+		let (_, pool) = Pools::<T>::iter().next().expect("Pool should exist");
+		assert_eq!(pool.state, PoolStatus::Active);
+	}
+
+	#[benchmark]
+	fn mint_into_polynomial(c: Linear<1, { <T as Config>::MaxCurrencies::get() }>) {
+		let origin = T::PoolCreateOrigin::try_successful_origin().expect("creating origin should not fail");
+		let account_origin = origin
+			.clone()
+			.into_signer()
+			.expect("generating account_id from origin should not fail");
+		make_free_for_deposit::<T>(&account_origin);
+
+		let collateral_id = create_default_collateral_asset::<T>();
+
+		let curve = get_linear_bonding_curve::<CurveParameterTypeOf<T>>();
+		let bonded_currencies = create_bonded_currencies_in_range::<T>(c, false);
+
+		let pool_id = create_pool::<T>(curve, bonded_currencies.clone(), None, None, None);
+
+		T::CollateralCurrencies::touch(collateral_id, &pool_id.clone().into(), &account_origin)
+			.expect("Touching should work");
+
+		let beneficiary = AccountIdLookupOf::<T>::from(account_origin.clone());
+		let amount_to_mint = 10u128;
+
+		#[extrinsic_call]
+		mint_into(
+			origin as T::RuntimeOrigin,
+			pool_id,
+			0,
+			beneficiary,
+			amount_to_mint.saturated_into(),
```

Comment:
No conversions in the benchmarked extrinsic itself, nor `::get()`. Same below in the other cases.

Reply:
[44841bd](https://github.com/KILTprotocol/kilt-node/pull/789/commits/44841bdaa050c2179d0748ac59fbf9357845197f)